tests/integration/synapseutils/test_synapseutils_sync.py

```diff
@@ -391,6 +368,311 @@ async def test_sync_to_synapse_activities_pointing_to_files(
+        assert len(folder.files[0].activity.used) == 1
+        assert folder.files[0].activity.used[0].url == SYNAPSE_URL
+
+        # WHEN I update the manifest file to remove the activities and update a metadata
+        # field
+        df = pd.DataFrame(
+            {
+                "path": [temp_file],
+                "parent": folder.id,
+                "contentType": "text/html",
+                "used": "",
+                "executed": "",
+                "activityName": "",
+                "activityDescription": "",
+            }
+        )
+        # Write the df to the file:
+        file_name = write_df_to_tsv(df, schedule_for_cleanup)
+
+        # AND I sync the manifest to Synapse
+        synapseutils.syncToSynapse(syn, file_name, sendMessages=SEND_MESSAGE, retries=2)
+
+        # THEN I expect that the folder has all of the files
+        await folder.sync_from_synapse_async(download_file=False)
+        assert len(folder.files) == 1
+
+        # AND the file is the one we uploaded
+        assert folder.files[0].version_number == 2
+        assert folder.files[0].path == temp_file
+        # AND none of the files have an activity
+        assert folder.files[0].activity is None
+
+        # AND the first version of the file still has the activity
+        first_file_version = await File(
+            id=folder.files[0].id, version_number=1
+        ).get_async(include_activity=True)
+        assert first_file_version is not None
+        assert first_file_version.activity is not None
+        assert first_file_version.activity.name == BOGUS_ACTIVITY
+        assert first_file_version.activity.description == BOGUS_DESCRIPTION
+        assert len(first_file_version.activity.used) == 1
+        assert first_file_version.activity.used[0].url == SYNAPSE_URL
+
+    async def test_sync_to_synapse_annotations_added_then_removed(
+        self, syn: Synapse, schedule_for_cleanup, project_model: Project
+    ) -> None:
+        """
+        Creates a number of files with annotations.
+        After the files are uploaded to Synapse, the annotations are removed from the
+        manifest file and the files are re-uploaded to Synapse. The annotations should
+        be removed from the files.
+        """
+        # GIVEN a folder to sync to
+        folder = await Folder(
+            name=str(uuid.uuid4()), parent_id=project_model.id
+        ).store_async()
+        schedule_for_cleanup(folder.id)
+
+        # AND 5 temporary files on disk:
+        temp_files = [utils.make_bogus_uuid_file() for _ in range(5)]
+        for file in temp_files:
+            schedule_for_cleanup(file)
+
+        # AND A manifest file with the paths to the temp files exists
+        annotations = ["foo", "bar", "baz", "qux", "quux"]
+        df = pd.DataFrame(
+            {
+                "path": temp_files,
+                "parent": folder.id,
+                "my_file_annotations": annotations,
```

Comment:
Is this existing behavior as well? This makes partial updates more difficult if you only want to update the annotations in one column.

Reply:
Excellent point. Perhaps the way I should make this work is you may specify optional booleans in the function:

1. Clear annotations from Synapse that are not defined in my file
2. Remove activity/provenance if not defined in my file
3. I could even allow something like: Apply existing provenance to new file versions

Reply:
That is a great idea.  For no. 3, we may want to hold off on that because technically when you create a new version of a file, you may have a new activity.  That said, maybe the new annotations added wouldn't have a new activity since the content in the file is the same.